sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -275,6 +275,103 @@ void gnrc_ndp_nbr_adv_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
     return;
 }
 
+void gnrc_ndp_rtr_adv_handle(kernel_pid_t iface, gnrc_pktsnip_t *pkt, ipv6_hdr_t *ipv6,
+                             ndp_rtr_adv_t *rtr_adv, size_t icmpv6_size)
+{
+    uint8_t *buf = (uint8_t *)(rtr_adv + 1);
+    gnrc_ipv6_nc_t *nc_entry = NULL;
+    gnrc_ipv6_netif_t *if_entry = gnrc_ipv6_netif_get(iface);
+    uint8_t l2src[GNRC_IPV6_NC_L2_ADDR_MAX];
+    int sicmpv6_size = (int)icmpv6_size, l2src_len = 0;
+    uint16_t opt_offset = 0;
+    assert(if_entry != NULL);
+    if (!ipv6_addr_is_link_local(&ipv6->src) ||
```

Comment:
Styling question, but for readability I prefer a newline between variable initializations and actual start of the functionality.